### ISSUE TITLE
Petitioning App

### ISSUE THREAD

**Issue by Woodley** [open-austin/project-ideas#61]
### Issues:

Paper petitions tend to be hard to read have limited space and have to be converted to electronic data. Not everyone has a smart phone, many have flip phones or like me, no phones. Many smart phones are too small for reading. Some people have vision issues. Many places people collect petitions do not have WiFi or other Internet access like Parks, Stores, Bus Stops, intersections, parking lots, park & rides, etc. just to name a few places. 

The app would only need to store information on a device until it connects to WiFi or USB to upload the collected information to the Cloud or server.

I was thinking that we could install these on a Kindle or Nook which can be seen in broad daylight and are easy to read in direct sunlight. We could remove the Kindle's restrictive OS and put something like CyanGenMod on it, but I only see Mods for Fire not the E-Ink or PaperWhite. Might have to use the Manufacturers KDK or SDK. On a Kindle or Nook we do not need color but if we allow others to use their iPad or Smart phones Color is nice. Expansion to support other devices now or later depending on time lines. Other Tablets could also be used to expand reach and usability.
### Design:

**Volunteer Section:**

Simple login with no password for volunteer to be able to say they collected petitions at a certain location, when first starting app. 
Have app remember last volunteer to log in and allow option to change before continuing to petition form. 
Link to Server Petition Sponsor provides.
Have volunteer user put in e-mail to get detailed report of petition they collected and submitted. 

**Petition Section:**

---

Petition Sponsor Name and/or Logo

---

First Name
Last Name
E-mail
Phone
Address
Zip
District #
Volunteer Yes/No

Please e-mail me confirmation

Sign me up for Newsletter.

Button Submit.

---

Information about Petition.

Sponsor Names and/or Logos.

---

**Additional Options:**

Auto date/time petition.

Needs to maintain a way to update app for new petitions in the future, By syncing with a Server to update information displayed like About Petition, Sponsor Names/Logo.

Secure Connections to Server.

In future; App might access Google Maps and GPS information for auto collecting locations but this can be added at a later date as it requires more time and effort.

I am open to other ideas.

At this time its not about making money off the app. If we want to make money off this app we could leave a link for credit (Giving created to builders and sponsors) that links to your other products and/or donation links. This would probably be the best option due to being a public servant tool. Otherwise a small fee for downloading from Google Play and Apple Store to pay for the exposure, Otherwise just have people install and update from a website where it is hosted. You do not have to use the Stores. Doesn't Google or Apple have breaks for non-profit tools?
### Current Petition:

A Petition I am currently supporting is through Bike Austin. Bike Austin is currently using this website, which I think needs a little improvement, but thats up to Bike Austin.
http://ourstreets.bikeaustin.org/

*If an App can not be completed in time for this petition drive it can be used for other petitions in the future.
### Who will benefit (directly and indirectly) from this project?

The Public, Sponsors, Volunteers, Non-profits, For-profit, Open Austin and Open Data. 
### Links to any research/data available/ articles

**Bike Austin:**
http://ourstreets.bikeaustin.org/

**Android Studio:**
http://developer.android.com/sdk/index.html

**Kindle Devolpment Kit (KDK)** for E-ink Readers, Paperwhite
http://kdk.amazon.com/gp/public/gateway

**Barnes & Nobles Nook (SDK)**
https://nookdeveloper.barnesandnoble.com/resources/index

**CyanGenMod:**
https://download.cyanogenmod.org/
### What are the next steps (validation, research, coding, design)?

All of the above.
### What help is needed at this time?

Design to Fruition.

**Comment by luqmaan** [comment]
Would something like https://www.change.org/petition meet your needs?

**Comment by Woodley** [comment]
> `luqmaan commented: 
Would something like https://www.change.org/petition meet your needs?`

@luqmaan 
No, That is not an App and it requires Internet Connection to fill out. Thats the same thing as the current Bike Austin Website has. I need something that does NOT require Internet while in the field collecting Names for petition digitally and can upload them when connected later.

**Comment by luqmaan** [comment]
https://itunes.apple.com/us/app/datafield-offline-form-survey/id470573506?mt=8

**Comment by Woodley** [comment]
```
> luqmaan commented: 
> https://itunes.apple.com/us/app/datafield-offline-form-survey/id470573506?mt=8
```
That is not a free app, It is a free trial, it looks very complicated and made for a wide range of customization. It cost a lot of money.
http://data-field.com/pricing/

Please do better research before replying.

**Comment by luqmaan** [comment]
https://itunes.apple.com/us/app/feed2go-create-conduct-free/id615209915?mt=8

**Comment by mateoclarke** [comment]
Hi @Woodley,

@luqmaan is just trying to help. Creating a custom app for Nook, Kindle, iOS, and Android would also be very costly. Developer's time is probably worth more than $90/month.

**Comment by Woodley** [comment]
```
>  luqmaan commented:
> https://itunes.apple.com/us/app/feed2go-create-conduct-free/id615209915?mt=8
```
@luqmaan
That looks like a survey app with in app purchases... where you formulate questions and graph responses. "All questionnaires can be obtained via an in-app purchase."
https://feed2go.com/use-cases/
It is also only available for iOS and iPad.

I have already tried to find something to collect petitions offline and could not. Good luck.

**Comment by Woodley** [comment]
```
> mateoclarke commented
> Creating a custom app for Nook, Kindle, iOS, and Android would also be very costly. Developer's time is probably worth more than $90/month.
```
@mateoclarke 
The point of this app is not about money but to be used for public service and expandable for any volunteer to use any where in the USA or over seas. Volunteers do not have money to spend on apps, they already donate time to a cause. Help make America better.

Credit for people that help create public service app should be enough. Donation links by the credit can be used to help pay for the app development and hosting. Credit can be used to direct people to your other projects that can collect income and expand, it can help your goals come to fruition. Think of this as an opportunity to get advertising exposure while making the world a better place.

**Comment by tonysms** [comment]
I'm a local Android dev and I'm willing to help / create the app. Can we meet and discuss the functionality?

**Comment by Woodley** [comment]
Hello Tony, I would be happy to meet to discuss this app. What part of the
day would be best for you and what area of Austin? I am in Far north Austin
but regularly go downtown for meetings. -John

**Comment by tonysms** [comment]
Hi Woodley,

I'm pretty flexible, otherwise weekend mornings work pretty much best 
for me for discussions & coffee :-)

I live @ the NW part of the town, so we can meet somewhere around Domain 
for example, or if you plan to attend the meetup this Saturday (Civic 
Hack Saturday ), we can meet there?

**Comment by Woodley** [comment]
I believe I will have time later this week in the afternoon or
evening. Weekends
are usually not good for me. maybe lunch at subway in the domain, I believe
they have wifi.

**Comment by tonysms** [comment]
NP. This Thursday or Friday afternoons work for me. Just let me know the 
place and time - you can contact me directly - tony at 
smartmobilesolutions dot com

**Comment by amaliebarras** [comment]
Hi @tonysms @Woodley ! Did y'all ever meet up to discuss this project? What's the current status?

**Comment by Woodley** [comment]
I appear to have missed this email. We did meet up. He has been very busy
with other projects and I have been very busy with other things. I am
contacting him directly to inquire about the status.

**Comment by mscarey** [comment]
I think this project is too difficult. The main feature request here is to use an e-ink device to run a complex app like a traditional tablet for vision/accessibility reasons. It's just beyond our ability to make that kind of device work that way.
